driver-core/src/main/com/mongodb/internal/connection/tlschannel/BufferAllocator.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
```

Comment:
All tlschannel code must be under the correct license - MIT even if we've modified the code. Unless we have permission from the (authors) to relicense the code.

Reply:
So you think I should add the MIT license to every file (the original code only contains the license in LICENSE.txt at the top level, and there is no copyright)?

Reply:
Added the license and copyright to every vendored file.